services/gif/gif.go

```diff
@@ -0,0 +1,71 @@
+package gif
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+)
+
+const maxRetry = 3
+const baseURL = "https://g.tenor.com/v1/"
+
+func NewGifAPI(db *accounts.Database) *API {
+	return &API{db, &http.Client{Timeout: time.Minute}}
+}
+
+// API is class with methods available over RPC.
+type API struct {
+	db         *accounts.Database
+	httpClient *http.Client
+}
+
+func (api *API) GetTenorAPIKey() (value string, err error) {
+	log.Info("[GifAPI::getTenorAPIKey]")
+	apiKey, err := api.db.GifAPIKey()
+	if err != nil {
+		return "", err
+	}
+	return apiKey, err
+}
+
+func (api *API) GetContentWithRetry(path string) (value string, err error) {
```

Comment:
That looks more like an internal API that should not be exposed. Could we have more dedicated endpoint?

Reply:
hmmm would something like getContent be better?

Reply:
Done @alaibe have also moved the recents and favourites here